### PR TITLE
Start building POWER10-optimized libraries

### PR DESCRIPTION
**Pull request by tuliom**
Notice this is still a WIP.

I'm seeing the following error when building GCC:

```
checking if mkdir takes one argument... no
Unknown cpu used in --with-cpu=power10.
make[2]: *** [Makefile:4338: configure-gcc] Error 1
make[2]: Leaving directory '/home/tuliom/tmp/at-build-tray/at14.0-0-alpha2.debian-10_ppc64le_ppc64le/builds/gcc_power10.tuned'
```

I've already confirmed the GCC detected at configure time does support `-mcpu=power10`.
It seems there is a configury code causing this error.
Needs further investigation.

### CODE REVIEW

**Comment by tuliom** [comment]
Aaron Sawdey explained to me this has been fixed on GCC master branch.
A backport is on the works for GCC 10.

https://gcc.gnu.org/git/?p=gcc.git;a=commitdiff;h=71237df0a0b7f0f10cebedcd114fae7ad2aaebcb

**Comment by tuliom** [comment]
I have rebased my previous PR and added a couple of fixes that I think will solve the issues we had found.
I've also removed the draft status from this PR.

**Comment by tuliom** [comment]
I forgot to mention that I'd like to hear comments if this GCC fix should also be backported to AT <= 12.0,

**Comment by er-1** [comment]
> I forgot to mention that I'd like to hear comments if this GCC fix should also be backported to AT <= 12.0,

@tuliom It could have some meaning to backport it to AT 12.0 as it'll be supported until mid 2021.

**Comment by tuliom** [comment]
> It could have some meaning to backport it to AT 12.0 as it'll be supported until mid 2021.

@er-1 Just confirming: I don't think the AT 12.0 changes must be made at the same time as this PR.  Do you agree?

**Comment by er-1** [comment]
> @er-1 Just confirming: I don't think the AT 12.0 changes must be made at the same time as this PR. Do you agree?

@tuliom I agree.

File: configs/14.0/packages/glibc/stage_optimized [hunk 1]
```diff
@@ -84,7 +84,8 @@ atcfg_configure() {
 	CXX="${at_dest}/bin/g++ -m${AT_BIT_SIZE}" \
 	CFLAGS="-g -O3 \
 		${with_longdouble:+-mlong-double-128} \
-		-fcommon" \
+		-fcommon \
+		-Wno-error=maybe-uninitialized" \
```
Comment (ThinkOpenly) [inline configs/14.0/packages/glibc/stage_optimized:88]: Is there an issue open to keep track of this?

**Comment by ThinkOpenly** [comment]
Also, shall we change the copyright year in configs/14.0/build.mk?

File: configs/14.0/packages/glibc/stage_optimized (hunk 1 above)
Comment (tuliom) [reply configs/14.0/packages/glibc/stage_optimized:88]: Done with issue #1635 .

**Comment by tuliom** [comment]
> Also, shall we change the copyright year in configs/14.0/build.mk?

@ThinkOpenly Done.

File: configs/14.0/packages/gcc/sources [hunk 2]
```diff
@@ -64,6 +64,12 @@ atsrc_get_patches ()
 		'https://raw.githubusercontent.com/powertechpreview/powertechpreview/master/GCC%20libstdc%2B%2B%20PowerPC%20Patches/9/0001-libstdc-Prevent-LD_LIBRARY_PATH-from-leaking-to-msgf.patch' \
 		9cba9a9b94ca9c1b8958c0d722a9c06b || return ${?}
 
+	# Fix missing notoc in libffi.
+	at_get_patch \
+		https://github.com/tuliom/gcc/commit/ba9a5fba9571bc7728b868bf8f812d7733488812.patch \
```
[the remark this answers is not in the document]
Comment (tuliom) [reply configs/14.0/packages/gcc/sources:69]: @mscastanho yes!  It's on my task list.

**Pull request merged**